common/util/src/test/java/org/thingsboard/common/util/ThingsBoardScheduledThreadPoolExecutorTest.java

```diff
@@ -0,0 +1,147 @@
+/**
+ * Copyright © 2016-2024 The Thingsboard Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.thingsboard.common.util;
+
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class ThingsBoardScheduledThreadPoolExecutorTest {
+
+    ThingsBoardScheduledThreadPoolExecutor scheduler;
+
+    @BeforeEach
+    void setup() {
+        scheduler = new ThingsBoardScheduledThreadPoolExecutor(1, Executors.defaultThreadFactory());
+    }
+
+    @AfterEach
+    void cleanup() {
+        scheduler.shutdownNow();
+    }
+
+    @Test
+    @DisplayName("scheduleAtFixedRate() should continue periodic execution even if command throws exception")
+    void scheduleAtFixedRateShouldNotStopPeriodicExecutionWhenCommandThrowsException() {
```

Comment:
I would like to see an example of start and stop scheduled task  to keep a normal lifecycle.
To add more context: see the DeviceProfileRuleNode that can be stopped and started many times in the rule engine due to rebalance or update. Or The rule node deletion

Reply:
![image](https://github.com/user-attachments/assets/9015bffc-ffe6-47ab-8534-d127ff0255ed)

Reply:
my suggestions here https://github.com/dskarzh/thingsboard/pull/1